[OSS] feat: add PeerThroughMeshGateways to mesh config (#14475)
hashicorp/consul: website/content/docs/connect/config-entries/mesh.mdx

```hcl
Kind      = "mesh"
Namespace = "default" # Can only be set to "default".

> Partition = "default"

Author: I got feedback from one of the engineers this was confusing, because not specifying anything will get you the right behavior for both OSS/ENT. Wanted to get your thoughts. That might also mean we don't need separate tabs for OSS and ENT.

Note that the Kubernetes example does not include a `partition` field. Configuration entries are applied on Kubernetes using [custom resource definitions (CRD)](/docs/k8s/crds), which can only be scoped to their own partition.

###  Peer Through Mesh Gateways

> Route peering control plane traffic through mesh gateways.

Reviewer: ```suggestion
Set the `PeerThroughMeshGateways` parameter to `true` to route peering control plane traffic through mesh gateways.
```
I think the indentations are causing the code blocks to render incorrectly.

<Tabs>
<Tab heading="Consul OSS">

<CodeTabs tabs={[ "HCL", "Kubernetes YAML", "JSON" ]}>

```hcl
Kind = "mesh"
Peering {
  PeerThroughMeshGateways = true
}
```

```yaml
apiVersion: consul.hashicorp.com/v1alpha1
kind: Mesh
metadata:
  name: mesh
spec:
  peering:
    peerThroughMeshGateways: true
```

```json
{
  "Kind": "mesh",
  "Peering": {
    "PeerThroughMeshGateways": true
  }
}
```

</CodeTabs>

</Tab>
<Tab heading="Consul Enterprise">

You can only set the `PeerThroughMeshGateways` attribute on `mesh` configuration entries in the `default` partition.

> This is the partition that owns mesh gateways routing control plane traffic to Consul servers.

Reviewer: ```suggestion
The `default` partition owns the traffic routed through the mesh gateway control plane to Consul servers.
```
Is that correct? I was having trouble parsing this sentence.

Author: You got it! 🙇

        },
      ],
    },
    {
      name: 'Peering',
      type: 'PeeringMeshConfig: <optional>',
      description:
        'Controls configuration specific to [peering connections](/docs/connect/cluster-peering).',
      children: [
        {
          name: 'PeerThroughMeshGateways',
          type: 'bool: <optional>',
          description: `Determines if peering control-plane traffic should be routed through mesh gateways.

>                         When enabled, dialing cluster will attempt to contact peers through their mesh gateway.
>                         Accepting clusters will advertise the address of their mesh gateways as opposed to their Consul servers directly.`,

Reviewer: ```suggestion
                        When enabled, dialing cluster attempt to contact peers through their mesh gateway.
                        Clusters that accept calls advertise the address of their mesh gateways, rather than the address of their Consul servers.`,
```
